generators/app/index.js

```diff
@@ -37,34 +37,70 @@ module.exports = class extends Generator {
     }
   }
 
-  async writing() {
-    this.destinationRoot(this.appPath);
-    this.templates.forEach(templatePath => {
-      let destinationPath =
-        templatePath == "gitignore" ? ".gitignore" : templatePath;
-      this.fs.copyTpl(
-        this.templatePath(templatePath),
-        this.destinationPath(destinationPath),
-        this.answers
-      );
-    });
+  writing() {
+    const templateDirectory = `elm/${this.answers.type}`;
+    const destDir = this.answers.name;
+
+    const props = {
+      name: this.answers.name,
+      installer: "yarn",
+      description: "",
+      author: "",
+      license: ""
+    };
+
+    this.fs.copyTpl(
+      this.templatePath(`${templateDirectory}/_elm.json`),
+      this.destinationPath(`${destDir}/elm.json`),
+      props
+    );
+
+    this.fs.copy(
+      this.templatePath(`${templateDirectory}/src`),
+      this.destinationPath(`${destDir}/src`)
+    );
+
+    //--- PARCEL
+    this.fs.copyTpl(
+      this.templatePath("parcel/style.css"),
+      this.destinationPath(`${destDir}/style.css`),
```

Comment:
I am still thinking about this one, so I'm just going to brain dump here and see if it helps. I'm hesitant to put js/css in a`/public` directory since I get the impression that people usually reserve `public` for things that aren't getting processed at all during the build, and parcel is definitely going to process all of those things, at least a little bit.  I looked around at some other scaffolding tools to see what they do, and mostly just got more confused.

Vue CLI creates this (some files removed for clarity):
```
vue
├── public
│   ├── favicon.ico
│   └── index.html
├── src
│   ├── App.vue
│   ├── assets
│   │   └── logo.png
│   ├── components
│   ├── main.ts
├── tests
│   └── unit
│       └── example.spec.ts
``` 
I look at that and think how is `logo.png` a source file? If a png goes in `src/` why don't `index.html` and `favicon.ico`?

Create elm app does this (also edited):
```
elm
├── public
│   ├── favicon.ico
│   ├── index.html
│   ├── logo.svg
│   └── manifest.json
├── README.md
├── src
│   ├── index.js
│   ├── main.css
│   ├── Main.elm
│   └── registerServiceWorker.js
└── tests
    └── Tests.elm
```
This makes a bit more sense to me. The things in `public/` actually seem pretty static, although you might eventually add build steps to process images and other files, in which case, do you move them to `src/`? 

I guess one thing I liked about having a single top level `app/` (or `src/`) with things under it like `elm/`, `js/`, `css/` (or `style/`) is that there's not much room for confusion about what goes where, since the folders are just named for the type of thing in them, not really for what's being done to them during a build/serve phase. 

An idea I like is that anybody who comes into an app created with `elm-app-gen` would know where to look for things because there's a standard setup. I do recognize your concern though about not including things we don't need.  I'm going to think about this one some more, maybe bounce some ideas off of other devs and come back to it in a day or two.

Reply:
After some reflexion i did the following : 

src
|____ Main.elm
|____ index.html
|____ css
           |____style.css
|____ js
           |____app.js

I agree with you that the public folder is not a good idea

Reply:
I think that's good for now. I spent some more time thinking about and talking to some other web-dev it and the only conclusion I came to was that `public` only makes sense if you've been doing web development for a while, and nobody can quite explain what it's for.